android/src/main/java/com/mattermost/networkclient/APIClientModule.kt

```diff
@@ -158,26 +228,26 @@ class APIClientModule(reactContext: ReactApplicationContext) : ReactContextBaseJ
 
         try {
             // Create a Request
-            var request = if(options?.hasKey("method") == true) {
+            var request = if (options?.hasKey("method") == true) {
                 when (options.getString("method")) {
-                    "PUT" -> sessionsRequest[baseUrl]!!.url(formUrlString(baseUrl, endpoint)).put(body)
-                    "PATCH" -> sessionsRequest[baseUrl]!!.url(formUrlString(baseUrl, endpoint)).patch(body)
+                    "PUT" -> Request.Builder().url(formUrlString(url, endpoint)).put(body)
+                    "PATCH" -> Request.Builder().url(formUrlString(url, endpoint)).patch(body)
                     // Default to "POST"
-                    else -> sessionsRequest[baseUrl]!!.url(formUrlString(baseUrl, endpoint)).post(body)
+                    else -> Request.Builder().url(formUrlString(url, endpoint)).post(body)
                 }
             } else {
-                sessionsRequest[baseUrl]!!.url(formUrlString(baseUrl, endpoint)).post(body)
+                Request.Builder().url(formUrlString(url, endpoint)).post(body)
```

Comment:
Might be a silly comment coming from outside but do we need GET method ?  I see we have put, patch and post but no get.

Reply:
I don't think you're supposed to send body data with a GET request; might be wrong - @enahum or @migbot thoughts?